gpflow/base.py

```diff
@@ -59,17 +60,35 @@ def __init__(self,
             self._unconstrained = tf.Variable(value, dtype=dtype, name=name, trainable=trainable)
 
         self.prior = prior
+        self.prior_on_constrained = prior_on_constrained
         self._transform = transform
 
-    def log_prior(self):
-        x = self.read_value()
-        y = self._unconstrained
+    def log_prior(self, evaluate_on_constrained: bool = True):
+        """ Prior probability density.
+        This can be evaluated either on the constrained or unconstrained variable.
+        For example if transform = Exp(), prior = Uniform(), then log_prior will either be
+        uniform (when evaluate_on_constrained is True), or scale as 1/value when
+        evaluate_on_constrained is set to False.
```

Comment:
This is specific to the `prior_on_constrained=True` setting (which is now the default), is that correct? Would it be worth mentioning that to be clear?

Reply:
Correct, the docstring is now updated to reflect this.